app/models/nomis/offender_base.rb

```diff
@@ -114,20 +114,16 @@ def load_from_json(payload)
       # It is expected that this method will be called by the subclass which
       # will have been given a payload at the class level, and will call this
       # method from it's own internal from_json
-      @first_name = payload['firstName']
-      @last_name = payload['lastName']
+      @first_name = payload.fetch('firstName')
+      @last_name = payload.fetch('lastName')
       @offender_no = payload.fetch('offenderNo')
       @convicted_status = payload['convictedStatus']
-      @sentence_type = SentenceType.new(payload['imprisonmentStatus'])
+      @sentence_type = SentenceType.new(payload['imprisonmentStatus'], payload['recall'])
```

Comment:
Part of me questions whether 'recall' status is really part of the 'sentence type', or whether that was just a convenient place to put it when we'd predicted recall status based on the sentence code.

It's a question of semantics – is the offender recalled, or are they serving a recall sentence? I think it could be the former, in which case I think it could benefit from being refactored into its own first-class attribute on `OffenderBase`.

Looking at the codebase, it seems like we always use `OffenderBase#recalled?`. We never directly call `SentenceType#recall_sentence?` except from in `OffenderBase#recalled?` when delegating. We're just passing the value of `recall` from `OffenderBase` into `SentenceType`, just so we can pull it back out again.

Reply:
My worry is that we already have too much code in OffenderBase. Its a bit of a GOD class pattern. I wish we had smaller simpler classes (e.g. Sentence?) for the cases when we don't take decisions based on things like first_name. SentenceType used to answer this question (in fact I have given it the immigration_case? method recently, OffenderBase was doing it). Here you're right, it's unclear which is better - especially as the API is answering the question directly

Reply:
I've gone with your suggestion. SentenceType is an anomaly causwed by us binding our data model to the NOMIS API - it's really an 'imprisonmentStatus decoder' which no longer has the recall? responsibility.